### PR TITLE
Feature resizable spilt with custom separator

### PR DESCRIPTION
**Pull request by ArthurSonzogni**
Create from: https://github.com/ArthurSonzogni/FTXUI/pull/581

This resolves:
https://github.com/ArthurSonzogni/FTXUI/issues/580

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
- I deduplicated the ResizableSplit class.
- I introduced ResizableSplitOptions, so that folks can use designated initializer to tweak the option they want.
- I stopped passing an Element, and replaced it by a function<Element()>. The reason is that an `Element` isn't meant to be used drawn on screen more than once.

I think it still requires some work. I would be very interested plumbing the mouse distance / position relative to the split to make some particular effects.

File: src/ftxui/component/resizable_split.cpp [hunk 1]
```diff
@@ -264,20 +148,24 @@ class ResizableSplitBottomBase : public ComponentBase {
 /// int left_size = 10;
 /// auto left = Renderer([] { return text("Left") | center;});
 /// auto right = Renderer([] { return text("right") | center;});
-/// auto split = ResizableSplitLeft(left, right, &left_size);
+/// auto split = ResizableSplitLeft(left, right, &left_size, separatorDouble());
```
Comment (pllee4) [inline src/ftxui/component/resizable_split.cpp:151]: minor: I guess can revert those changes on interface which was "polluted" by me for `ResizableSplit...` for the example code in the documentation.

**Comment by pllee4** [comment]
Looks great for the changes, thanks!

**Comment by ArthurSonzogni** [comment]
Sorry for the delay. I wanted to add a way to visualize the mouse was close to the resizeableSplit, by passing kind of coordinates to the function. I think I will find another way in the future, or maybe break the current API.

**Pull request merged**